data/npc/scripts/bank.lua

```diff
@@ -0,0 +1,318 @@
+			npcHandler.topic[cid] = 0
+		end
+		return true
+	elseif msgcontains(msg, "transfer") then
+		npcHandler:say("Please tell me the amount of gold you would like to transfer.", cid)
+		npcHandler.topic[cid] = 11
+	elseif npcHandler.topic[cid] == 11 then
+		count[cid] = getMoneyCount(msg)
+		if player:getBankBalance() < count[cid] then
+			npcHandler:say("There is not enough gold in your account.", cid)
+			npcHandler.topic[cid] = 0
+			return true
+		end
+		if isValidMoney(count[cid]) then
+			npcHandler:say("Who would you like transfer " .. count[cid] .. " gold to?", cid)
+			npcHandler.topic[cid] = 12
+		else
+			npcHandler:say("There is not enough gold in your account.", cid)
+			npcHandler.topic[cid] = 0
+		end
+	elseif npcHandler.topic[cid] == 12 then
+		transfer[cid] = msg
+		if player:getName() == transfer[cid] then
+			npcHandler:say("Fill in this field with person who receives your gold!", cid)
+			npcHandler.topic[cid] = 0
+			return true
+		end
+		if playerExists(transfer[cid]) then
+			local transferPlayer = Player(transfer[cid])
+            local currentVocation = transferPlayer and transferPlayer:getVocation():getId() or getPlayerVocationAux(transfer[cid])
+            if currentVocation == VOCATION_NONE then
+                npcHandler:say("I'm afraid this character only holds a junior account at our bank. Do not worry, though. Once he has chosen his vocation, his account will be upgraded.", cid)
+                npcHandler.topic[cid] = 0
+				return true
+            end
+			npcHandler:say("So you would like to transfer " .. count[cid] .. " gold to " .. transfer[cid] .. "?", cid)
+			npcHandler.topic[cid] = 13
+		else
+			npcHandler:say("This player does not exist.", cid)
+			npcHandler.topic[cid] = 0
+		end
+	elseif npcHandler.topic[cid] == 13 then
+		if msgcontains(msg, "yes") then
+			if not player:transferMoneyTo(transfer[cid], count[cid]) then
+				npcHandler:say("You cannot transfer money to this account.", cid)
+			else
+				npcHandler:say("Very well. You have transfered " .. count[cid] .. " gold to " .. transfer[cid] ..".", cid)
+				transfer[cid] = nil
+			end
+		elseif msgcontains(msg, "no") then
+			npcHandler:say("Alright, is there something else I can do for you?", cid)
+		end
+		npcHandler.topic[cid] = 0
+	elseif msgcontains(msg, "change gold") then
+		npcHandler:say("How many platinum coins would you like to get?", cid)
+		npcHandler.topic[cid] = 14
+	elseif npcHandler.topic[cid] == 14 then
+		if getMoneyCount(msg) < 1 then
+			npcHandler:say("Sorry, you do not have enough gold coins.", cid)
+			npcHandler.topic[cid] = 0
+		else
+			count[cid] = getMoneyCount(msg)
+			npcHandler:say("So you would like me to change " .. count[cid] * 100 .. " of your gold coins into " .. count[cid] .. " platinum coins?", cid)
+			npcHandler.topic[cid] = 15
+		end
+	elseif npcHandler.topic[cid] == 15 then
+		if msgcontains(msg, "yes") then
+			if player:removeItem(2148, count[cid] * 100) then
```

Comment:
use the constants:
	ITEM_GOLD_COIN = 2148,
	ITEM_PLATINUM_COIN = 2152,
	ITEM_CRYSTAL_COIN = 2160,